sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FluxByteBufferContent.java

```diff
@@ -125,31 +124,47 @@ public BinaryDataContent toReplayableContent() {
             return replayableContent;
         }
 
-        Flux<ByteBuffer> bufferedFlux = content
-            .map(buffer -> {
-                // deep copy direct buffers
-                ByteBuffer copy = ByteBuffer.allocate(buffer.remaining());
-                copy.put(buffer);
-                copy.flip();
-                return copy;
-            })
-            // collectList() uses ArrayList. We don't want to be bound by array capacity
-            // and we don't need random access.
-            .collect(LinkedList::new, (BiConsumer<LinkedList<ByteBuffer>, ByteBuffer>) LinkedList::add)
-            .cache()
-            .flatMapMany(
-                // Duplicate buffers on re-subscription.
-                listOfBuffers -> Flux.fromIterable(listOfBuffers).map(ByteBuffer::duplicate));
-        replayableContent = new FluxByteBufferContent(bufferedFlux, length, true);
-        cachedReplayableContent.set(replayableContent);
-        return replayableContent;
+        return bufferContent().map(bufferedData -> {
+            FluxByteBufferContent bufferedContent = new FluxByteBufferContent(Flux.fromIterable(bufferedData)
+                .map(ByteBuffer::duplicate), length, true);
+            cachedReplayableContent.set(bufferedContent);
+
+            return bufferedContent;
+        }).block();
```

Comment:
Should we consider using `publishOn` to use `boundedElastic` scheduler?

Reply:
Maybe, we didn't before so this doesn't change that behavior, so we may want to put that off until next release.

The `publishOn` would get the handling of the `Flux` off any non-blocking thread, such as the Netty IO threads, right? And this could help more quickly free up those threads to take on another network request?

Reply:
Given this retains previous behavior, this will be held off on for now and reevaluated later.